components/AlertsDashboard.vue

```diff
@@ -165,7 +169,61 @@ onMounted(() => {
           );
           map.value.flyTo({ center: [lng, lat], zoom: 13 });
         }
+        isMapeo.value = false;
       }
+    } else if (mapeoDocId && props.mapeoData) {
+      // Wait for the next tick to ensure mapeo-data layer is added
+      nextTick(() => {
+        if (!props.mapeoData) {
+          return;
+        }
+
+        const mapeoFeature = props.mapeoData.find((f) => f.ID === mapeoDocId);
+
+        if (mapeoFeature) {
+          const geometryType = mapeoFeature.geotype as
+            | "Polygon"
+            | "LineString"
+            | "Point"
+            | "MultiPoint"
+            | "MultiLineString"
+            | "MultiPolygon"
+            | "GeometryCollection";
+          const feature: Feature = {
+            type: "Feature",
+            geometry: {
+              type: geometryType,
+              coordinates: JSON.parse(mapeoFeature.geocoordinates),
+            } as Geometry,
+            properties: {
+              ...mapeoFeature,
+            },
+          };
+
+          selectFeature(feature, "mapeo-data");
```

Comment:
In runtime, this throws an error and the Mapeo feature is not selected (e.g. is highlighted yellow on the map):

```
Error: The source 'mapeo-data' does not exist in the map's style.
```

The reason is that there is a race condition issue happening. 

On mount, we call `prepareMapCanvasContent()` which adds all of the alerts and Mapeo data to the map. But presently we don't wait for that to finish, and consequently, by the time the code that you've added is run, there is no guarantee that the data has been added to the map layer to be able to be selected.

So we'll need to turn `prepareMapCanvasContent()` into an async function and return a promise, probably best done using `Promise.all()`.